src/Adapter/MSTest.TestAdapter/ObjectModel/UnitTestElement.cs

```diff
@@ -249,9 +281,26 @@ internal TestCase ToTestCase()
             idProvider.AppendString(testCase.DisplayName);
         }
 
-        testCase.Id = idProvider.GetId();
+        return idProvider.GetId();
+    }
 
-        return testCase;
+    private Guid GenerateSerializedDataStrategyTestId()
+    {
+        var idProvider = new TestIdProvider();
+        idProvider.AppendString(Constants.ExecutorUriString);
+        idProvider.AppendString(TestMethod.AssemblyName);
```

Comment:
It's not fully clear to me if this is important to keep or not. I was told that we are supposed to consider 2 assemblies in different locations as different (this would help to disambiguate metadata for example) but we are not doing that in legacy nor display name (we only look at assembly name not assembly path).

@Haplois Do you know if that's a requirement?

Reply:
There is a bug there, trying to fix it :)

Reply:
Do you mean this is a bug on older versions and this is something we want to fix with new strategy?

Reply:
Fix is attached.

[0001-Fixed-testId-generation-bug.zip](https://github.com/microsoft/testfx/files/9869774/0001-Fixed-testId-generation-bug.zip)

Reply:
Few questions:
- What's the problem with using `AssemblyName` for path?
- You are using `Path.GetFileName` which gives only assembly name and not full path, is it expected?
- What's the problem with relying only on `FullClassName`? And not the ManagedXXX?

Reply:
We want to migrate away from FullyQualifedName and stop using it since it's not standardized. 

First two questions have the same answer: backwards compatibility. This is how TP handles ID gen.

Reply:
Ok but if that's backward compatibility it's ok to not use that in the new strategy because we don't have to care about past. As for `FullyQualifiedName`, I haven't seen any case where I have a value that differ so I am not sure to really understand the benefit.

The fact that it is not standard might be bothering for display name, UI positioning or something else but for test ID I don't see any problem. Also, looking at how it's done, we use runtime reflection info so I don't see how it could be bad/wrong.